src/Controller/Controller.php

```diff
@@ -410,6 +517,31 @@ public function setRequest(ServerRequest $request)
         if ($request->getParam('pass')) {
             $this->passedArgs = $request->getParam('pass');
         }
+
+        return $this;
+    }
+
+    /**
+     * Gets the response instance.
+     *
+     * @return \Cake\Http\Response
+     */
+    public function getResponse()
+    {
+        return $this->response;
+    }
+
+    /**
+     * Sets the response instance.
+     *
+     * @param \Cake\Http\Response $response Response instance.
+     * @return $this
+     */
+    public function setResponse(Response $response)
+    {
+        $this->response = $response;
```

Comment:
Because responses and requests are immutable I'm wondering if we should start thinking of ways to make working with responses less clunky. Using setResponse() is going to get annoying.